File: src/main/java/Electro2D/HTMLSorter.java
package main.java.Electro2D;/*
 * Copyright (C) 2013 Rochester Institute of Technology
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License as
 * published by the Free Software Foundation; either version 2 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 *
 * See the GNU General Public License for more details.
 */

import java.util.TreeSet;

/**
 * This class creates the ordered Vector of all of the proteins to be
 * displayed in hte HTML file.
 */

import java.util.Vector;

public class HTMLSorter {

    private TreeSet<Vector<String>> proteinInfo;
    private int compBy;

    public HTMLSorter(int cb, Vector<String> t, Vector<String> p, Vector<String> m, Vector<String> f) {

        compBy = cb;
        proteinInfo = new TreeSet<Vector<String>>(new HTMLComparator(compBy));
        Vector<String> tmp;

        for (int i = 0; i < t.size(); i++) {
            tmp = new Vector<String>();
            tmp.add(t.elementAt(i));
            tmp.add(p.elementAt(i));
            tmp.add(m.elementAt(i));
            tmp.add(f.elementAt(i));
            proteinInfo.add(tmp);
        }
    }

    public TreeSet<Vector<String>> getSorted() {
        return proteinInfo;
    }
}

File: site/swingjs/j2s/main/java/Electro2D/HTMLSorter.js
(function(){var P$=Clazz.newPackage("main.java.Electro2D"),I$=[[0,'java.util.TreeSet','main.java.Electro2D.HTMLComparator','java.util.Vector']],I$0=I$[0],$I$=function(i,n){return((i=(I$[i]||(I$[i]=Clazz.load(I$0[i])))),!n&&i.$load$&&Clazz.load(i,2),i)};
/*c*/var C$=Clazz.newClass(P$, "HTMLSorter");

C$.$clinit$=2;

Clazz.newMeth(C$, '$init$', function () {
},1);

C$.$fields$=[['I',['compBy'],'O',['proteinInfo','java.util.TreeSet']]]

Clazz.newMeth(C$, 'c$$I$java_util_Vector$java_util_Vector$java_util_Vector$java_util_Vector',  function (cb, t, p, m, f) {
;C$.$init$.apply(this);
this.compBy=cb;
this.proteinInfo=Clazz.new_([Clazz.new_($I$(2,1).c$$I,[this.compBy])],$I$(1,1).c$$java_util_Comparator);
var tmp;
for (var i=0; i < t.size$(); i++) {
tmp=Clazz.new_($I$(3,1));
tmp.add$O(t.elementAt$I(i));
tmp.add$O(p.elementAt$I(i));
tmp.add$O(m.elementAt$I(i));
tmp.add$O(f.elementAt$I(i));
this.proteinInfo.add$O(tmp);
}
}, 1);

Clazz.newMeth(C$, 'getSorted$',  function () {
return this.proteinInfo;
});

Clazz.newMeth(C$);
})();
;Clazz.setTVer('3.3.1-v4');//Created 2022-04-21 16:53:54 Java2ScriptVisitor version 3.3.1-v4 net.sf.j2s.core.jar version 3.3.1-v4
